### PR TITLE
En passant returns the square only if there is a pawn that could capture

### PR DESCRIPTION
**Pull request by Jilyfe**
Should check wether the pawn can actually capture (not pinned)

### CODE REVIEW

**Comment by programarivm** [comment]
Thank you @Jilyfe for sending this PR, it is very much appreciated! 

It'd be helpful if you could also provide a FEN example with the bug.

Keep it up,

**Comment by Jilyfe** [comment]
Here is an example, we get 2 boards we the exact same position (since there is no pawn that can capture en passant), but the fen we get is different.

`

        use Chess\Variant\Classical\Board;

        $board1 = new Board();
        $board2 = new Board();
        $moves1 = ['d4', 'd5', 'Nf3'];
        $moves2 = ['Nf3', 'd5', 'd4'];

        foreach ($moves1 as $move) {
            $board1->play($board1->turn, $move);
        }
        foreach ($moves2 as $move) {
            $board2->play($board2->turn, $move);
        }

        var_dump($board1->toFen()); // rnbqkbnr/ppp1pppp/8/3p4/3P4/5N2/PPP1PPPP/RNBQKB1R b KQkq -
        var_dump($board2->toFen()); // rnbqkbnr/ppp1pppp/8/3p4/3P4/5N2/PPP1PPPP/RNBQKB1R b KQkq d3`

**Comment by programarivm** [comment]
Excellent, thanks!

PHP Chess is currently sticking to the spec as described at [En passant target square](https://www.chessprogramming.org/Forsyth-Edwards_Notation#En_passant_target_square):

> The en passant target square **is specified after a double push of a pawn, no matter whether an en passant capture is really possible or not**. Other moves than double pawn pushes imply the symbol '-' for this FEN field.

The implementation guide at [16.1.3.4: En passant target square](https://www.saremba.de/chessgml/standards/pgn/pgn-complete.htm#c16.1.3) is perhaps not that accurate when describing the en passant target square, but it still says:

> An en passant target square is given if and only if the last move was a pawn advance of two squares. Therefore, an en passant target square field **may have a square name even if there is no pawn of the opposing side** that may immediately execute the en passant capture.

In conclusion, since the en passant target square is always specified after a double push of a pawn, `1.♘f3 d5 2.d4` has a square name indeed even though there is no pawn of the opposing side that may immediately execute the en passant capture.

May I ask where the example came from?
